src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -634,6 +652,9 @@ private void processLogsAndUpload() throws InterruptedException {
 
                             if (startPosition < file.length()) {
                                 unitOfWork.getLogFileInformationList().add(logFileInformation);
+                            } else if (startPosition == file.length() && !logFileGroup.isActiveFile(file)) {
+                                updatelastComponentUploadedLogFile(lastComponentUploadedLogFileInstantMap,
+                                                                    componentName, file.lastModified());
```

Comment:
Here is the fix of the data duplicate bugs. Here is some summaries: 
1. In the first scan: When the active file (with hash1) is fully written and processed but not rotated, the startPos will be sat as the end of the file. After uploading, this file is saved as currentProcessing file
2. In the second scan: this file (with hash1) is rotated but still can be picked up the right startPos because it is marked as the currentProcessing file. But when we process the file, we will skip it because startPos == file.length(). After uploading, the currentProcessing file will be pointed to a new file.
3. In the third scan: the file (with hash1) will be picked up again, because the lastUploaded file didn`t update, but this time, we cannot sync the startPos. Then duplicate happens. 
 
Why this fix the bug: 
Once we figure out this file (with hash1) is rotated and startPos == file.length, we treat it as the lastUploaded file, so we try to update the global value. This is before we skip it in the processing steps.